pkg/nsx/services/vpc/store.go

```diff
@@ -74,28 +78,25 @@ func (vs *VPCStore) Apply(i interface{}) error {
 	return nil
 }
 
-func (vs *VPCStore) GetVPCsByNamespace(ns string) []*model.Vpc {
-	var ret []*model.Vpc
-	vpcs := vs.List()
-	if len(vpcs) == 0 {
-		log.V(1).Info("No vpc found in vpc store")
-		return ret
+func (s *VPCStore) GetByIndex(key string, value string) []*model.Vpc {
+	vpcs := make([]*model.Vpc, 0)
+	objs := s.ResourceStore.GetByIndex(key, value)
+	for _, vpc := range objs {
+		vpcs = append(vpcs, vpc.(*model.Vpc))
 	}
+	return vpcs
+}
 
-	for _, vpc := range vpcs {
-		mvpc := vpc.(*model.Vpc)
-		tags := mvpc.Tags
-		for _, tag := range tags {
-			if *tag.Scope == common.TagScopeNamespace && *tag.Tag == ns {
-				ret = append(ret, mvpc)
-			}
-		}
-	}
-	return ret
+func (s *VPCStore) GetVPCsByNamespace(ns string) []*model.Vpc {
+	return s.GetByIndex(common.TagScopeNamespace, ns)
+}
+
+func (s *VPCStore) GetVPCsByNamespaceID(namespaceID string) []*model.Vpc {
```

Comment:
I'm doubting the use case for  `GetVPCsByNamespaceID`, since usually in fetching stale VPCs case, we need to list all VPS associated with ns name not ns ID. but it's not be harmful to put here at this moment.

Reply:
Is it related to this PR change? The `GetVPCsByNamespaceID` is used in creating a Subnet, since the VPCs fetched by Namespace ID must not be stale.

Reply:
The VPCs fetched by Namespace name may be stale.